File: backJoon/src/Q2309_일곱난쟁이/Main.java
package Q2309_일곱난쟁이;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class Main {
	static List<Integer> list;
	static int sum;
	static int[] arr;
	static boolean[] visited;
	static int len;

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		arr = new int[9];
		sum = 0;
		len = arr.length;

		for (int i = 0; i < arr.length; i++) {
			arr[i] = sc.nextInt();
			sum += arr[i];
		}

		visited = new boolean[arr.length];

		comb(0, 2);
	}

	static void comb(int st, int r) {
		if (r == 0) {
			int nonD = 0;
			for (int i = 0; i < len; i++) {
				if (visited[i]) {
					nonD += arr[i];
				}
			}

			if (sum - nonD == 100) {
				check();
				for (int l : list) {
					System.out.println(l);
				}
				System.exit(0);
			}
			return;
		}

		for (int i = 0; i < len; i++) {
			if (!visited[i]) {
				visited[i] = true;
				comb(i + 1, r - 1);
				visited[i] = false;
			}
		}
	}

	static void check() {
		list = new ArrayList<>();
		for (int i = 0; i < len; i++) {
			if (!visited[i]) {
				list.add(arr[i]);
			}
		}
		Collections.sort(list);
	}
}

File: backJoon/src/Q2309_일곱난쟁이/main.cpp
#include <bits/stdc++.h>
using namespace std;
int arr[9];
vector<int> temp;

void printV() {
    for(int i=0; i<temp.size(); i++) {
        cout << temp[i] << "\n"; 
    }
    exit(0);
}

void comb(int d, int n, int r, int idx) {
    if(d>=r) {
        if(accumulate(temp.begin(), temp.end(), 0)==100) printV();
        return;
    }
        
    for(int i=idx; i<n; i++) {
        temp.push_back(arr[i]);
        comb(d+1, n,r,idx+1);
        temp.pop_back();
    }
}

int main() {
    for(int i=0; i<9; i++) {
        cin >> arr[i];
    }
    sort(arr, arr+9);
    comb(0,9,7,0);
    return 0; 
}